fhttp/httprunner_test.go

```diff
@@ -471,6 +472,107 @@ func TestAbortOn(t *testing.T) {
 	}
 }
 
+func TestConnectionReuseRange(t *testing.T) {
+	mux, addr := DynamicHTTPServer(false)
+	mux.HandleFunc("/foo/", EchoHandler)
+	url := fmt.Sprintf("http://localhost:%d/foo/", addr.Port)
+	opts := HTTPRunnerOptions{}
+	opts.Init(url)
+	opts.QPS = 50
+	opts.URL = url
+	opts.NumThreads = 1
+	opts.Exactly = 10
+
+	// Test empty reuse range.
+	res, err := RunHTTPTest(&opts)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if res.SocketCount != 1 {
+		t.Errorf("Expected only 1 socket should be used when the max connection reuse flag is not set.")
+	}
+
+	// Test connection reuse range from 1 to 10.
+	for i := 1; i <= 10; i++ {
+		opts.ConnReuseRange = [2]int{i, i}
+		expectedSocketReuse := math.Ceil(float64(opts.Exactly) / float64(opts.ConnReuseRange[0]))
+		res, err := RunHTTPTest(&opts)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if res.SocketCount != (int)(expectedSocketReuse) {
+			t.Errorf("Expecting %f socket to be used, got %d", expectedSocketReuse, res.SocketCount)
+		}
+	}
+
+	// Test when connection reuse range min != max.
+	// The actual socket count should always be 2 as the connection reuse range varies between 5 and 9.
+	expectedSocketReuse := 2.0
+	for i := 0; i < 5; i++ {
+		opts.ConnReuseRange = [2]int{5, 9}
```

Comment:
move out of the loop as it's not changing?

the loop is here just because of rand right?